Add getting started and versioning docs (#505)
adobe-private/react-spectrum-v3: packages/dev/docs/pages/react-stately/getting-started.mdx

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. -->

import {Layout} from '@react-spectrum/docs';
export default Layout;

---
category: Introduction
---

# Getting Started

This page describes how to get started building your own component library with React Stately.

## What is React Stately?

**React Stately** is library of React Hooks that provides cross-platform state management for your design system.
It provides the foundation and core logic for your component library, and implements state management for many
common components. It returns an interface for reading and updating component state, with methods that implement
much of the core logic for the component. 

React Stately is designed to be cross-platform, and could work on the web, react-native, or any other platform.
On the web, it can be paired with [React Aria](../react-aria/index.html), which provides behavior, accessibility,

> and user interactions for many components, and uses React Stately to handle state management.

Reviewer: ```suggestion
and user interactions for many components, while React Stately provides state management.
```

## Installation

React Stately is **incrementally adoptable**. Each component is published as a separate package, so you can try it
out in a single component and gradually add more over time. All of these packages are published under the 
[@react-stately](https://www.npmjs.com/org/react-stately) scope on npm.

> You can install any React Aria package using a package manager like [npm](https://docs.npmjs.com/cli/npm) or 

Reviewer: ```suggestion
You can install any React Stately package using a package manager like [npm](https://docs.npmjs.com/cli/npm) or 
```

```
yarn add @react-stately/radio
```

## Building a component

This example shows a very simple radio group component built with the [useRadioGroupState](useRadioGroupState.html) hook
to manage state. It generates a name for each input in the group so the browser associates them, and manages the
selection state. The example shows [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html) behavior 
using the `defaultValue` prop, and `useRadioGroupState` handles firing the `onChange` event with the new selected value.

```tsx example
import {useRadioGroupState} from '@react-stately/radio';

function RadioGroup(props) {
  let state = useRadioGroupState(props);

  return (
    <>
      <label>
        <input 
          type="radio"
          name={state.name}
          checked={state.selectedValue === 'dogs'}
          onChange={() => state.setSelectedValue('dogs')} />
        Dogs
      </label>
      <label>
        <input 
          type="radio"
          name={state.name}
          checked={state.selectedValue === 'cats'}
          onChange={() => state.setSelectedValue('cats')} />
        Cats
      </label>
    </>
  );
}

<RadioGroup
  defaultValue="dogs"

>   onChange={value => alert(`Selected ${value}`)} />

Author: What do you think of this example, which uses stately without aria?

Reviewer 2: I think it is fine as is since you have a callout beneath it referencing react aria should the user want behavior + accessibility properly handled. The example in that doc combines the RadioGroup stately and aria hooks so no need to cover it in this example (a direct link + callout to the example in the useRadioGroup page would be a plus IMO but not necessary).
 If we are worried that users will ignore aria and only use stately perhaps a cautionary line should be placed after the sentence below stating something along the lines of "If you choose not to use the react-aria package, note that you will have to handle behavior/accessibility yourself".

```

> If you're using React Stately on the web, see [useRadioGroup](../react-aria/useRadioGroup.html) in React Aria to help

Reviewer: I would add a link to a more involved stately package here as well. Something like "To see an example of how React Stately manages state for more complex components such as selection, see our [useMultipleSelectionState] docs.
